src/plonk.rs

```diff
@@ -130,9 +131,29 @@ impl<C: HaloCurve> Circuit<C> {
             &plonk_z_coeffs,
         );
 
+        if cfg!(debug_assertions) {
+            // Check that the vanishing polynomial indeed vanishes.
+            self.subgroup_n.iter().for_each(|&x| {
+                assert!(eval_poly(&vanishing_coeffs, x).is_zero());
+            });
+        }
+
         // Compute the quotient polynomial, t(x) = vanishing(x) / Z_H(x).
-        let plonk_t_coeffs: Vec<C::ScalarField> = divide_by_z_h(&vanishing_coeffs, self.degree());
+        let mut plonk_t_coeffs: Vec<C::ScalarField> = divide_by_z_h(&vanishing_coeffs, self.degree());
+
+        if cfg!(debug_assertions) {
+            // Check that division was performed correctly by evaluating at a random point.
+            let xxx = C::ScalarField::rand();
+            assert_eq!(
+                eval_poly(&plonk_t_coeffs, xxx),
+                eval_poly(&vanishing_coeffs, xxx) / eval_zero_poly(self.degree(), xxx)
+            );
+        }
 
+        // Pad the coeffiecients to length a multiple of the degree.
```

Comment:
Hm, I thought it would always have degree `QUOTIENT_POLYNOMIAL_DEGREE_MULTIPLIER * self.degree()`. Is it the zero-trimming in `divide_by_z_h` that makes that not the case?

If it isn't already guaranteed, I think technically we should pad to that degree in particular, since the recursive circuit expects `QUOTIENT_POLYNOMIAL_DEGREE_MULTIPLIER` chunks. (Though maybe it wouldn't come up in practice.)

Reply:
Yes `divide_by_z_h` removes all trailing zero coefficients.
Do you mean changing the current check to something like:
```rust
if plonk_t_coeffs.len() != QUOTIENT_POLYNOMIAL_DEGREE_MULTIPLIER * self.degree() {
            plonk_t_coeffs.extend((0..QUOTIENT_POLYNOMIAL_DEGREE_MULTIPLIER * self.degree()-plonk_t_coeffs.len()).map(|_| C::ScalarField::ZERO));
        }
```
?

Reply:
Yeah I think that makes sense, or to tweak it slightly, the range could be `plonk_t_coeffs.len()..QUOTIENT_POLYNOMIAL_DEGREE_MULTIPLIER * self.degree()`

I suspect the prover may be a tad faster if we skip zero-trimming, since these polynomials would be pretty random looking, so we'd probably never trim down to a smaller power of two (except maybe for trivial circuits). Not sure if the time used for trimming+padding is significant; I could do a comparison after this PR lands.